### PR TITLE
doc: remove mention of X11 license variant

### PR DESCRIPTION
**Pull request by asymmetric**
The Nixpkgs license is pure MIT.

Cf. https://en.wikipedia.org/wiki/MIT_License#X11_License

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by fricklerhandwerk — APPROVED** [review]

**Comment by asymmetric** [comment]
@FRidh it seems you introduced the "permissive MIT/X11 license" formulation in 92179785b60d9fd9e67ac5301e78b892b1b7a0c7, but the [X11 clause](https://github.com/NixOS/nixpkgs/blob/92179785b60d9fd9e67ac5301e78b892b1b7a0c7/COPYING) is not present in the license (it had a [different addition](https://github.com/NixOS/nixpkgs/blob/92179785b60d9fd9e67ac5301e78b892b1b7a0c7/COPYING), but that doesn't seem related to the X11 clause). Was this an oversight, and do you agree with removing reference to X11?

/cc @alyssais as per @roberth's suggestion

**Comment by asymmetric** [comment]
Also /cc @edolstra as the one who added the COPYING file way back [in 2006](6b0b4586945e3531a04561f4fcffd7e5f666513d): besides the notice that was removed [in this commit](ef54c618dc7af978e875704b0142d3d1e8b96f29), the intention was for Nixpkgs to be pure-MIT, no X11 variant, right?

File: doc/preface.chapter.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 The Nix Packages collection (Nixpkgs) is a set of thousands of packages for the
 [Nix package manager](https://nixos.org/nix/), released under a
-[permissive MIT/X11 license](https://github.com/NixOS/nixpkgs/blob/master/COPYING).
+[permissive MIT license](https://github.com/NixOS/nixpkgs/blob/master/COPYING).
```
Comment (alyssais) [inline doc/preface.chapter.md:5]: We could also call it `MIT/Expat` if we want to keep this level of specificity.

**Review by alyssais — APPROVED** [review]

**Comment by FRidh** [comment]
> @FRidh it seems you introduced the "permissive MIT/X11 license" formulation in [9217978](https://github.com/NixOS/nixpkgs/commit/92179785b60d9fd9e67ac5301e78b892b1b7a0c7), but the [X11 clause](https://github.com/NixOS/nixpkgs/blob/92179785b60d9fd9e67ac5301e78b892b1b7a0c7/COPYING) is not present in the license (it had a [different addition](https://github.com/NixOS/nixpkgs/blob/92179785b60d9fd9e67ac5301e78b892b1b7a0c7/COPYING), but that doesn't seem related to the X11 clause). Was this an oversight, and do you agree with removing reference to X11?
> 
> /cc @alyssais as per @roberth's suggestion

Probably, and I suppose yes.

**Pull request merged**